scripts/superdesk/editor2/add-content.js

```diff
@@ -162,7 +165,24 @@ function AddContentCtrl (scope, element, superdesk, editor, $timeout, config) {
                     scope.node.focus();
                     vm.textBlockCtrl.restoreSelection();
                 });
-            }
+            },
+            pasteBlock: function() {
+                if (!vm.sdEditorCtrl.getCutBlock()) {
+                    return false;
+                }
+                var indexWhereToAddNewBlock = vm.sdEditorCtrl.getBlockPosition(vm.textBlockCtrl.block) + 1;
+                // cut the text that is after the caret in the block and save it in order to add it after the embed later
+                var textThatWasAfterCaret = vm.textBlockCtrl.extractEndOfBlock().innerHTML;
+                if (textThatWasAfterCaret && textThatWasAfterCaret !== '') {
```

Comment:
You can just say `if (textThatWasAfterCaret)` and it will cover the second assertion too. An empty string is a falsy value.